relay-general/src/pii/compiledconfig.rs

```diff
@@ -0,0 +1,142 @@
+use std::cmp::Ordering;
+use std::collections::BTreeSet;
+
+use crate::pii::builtin::BUILTIN_RULES_MAP;
+use crate::pii::{PiiConfig, Redaction, RuleSpec, RuleType};
+use crate::processor::SelectorSpec;
+
+/// A representation of `PiiConfig` that is more (CPU-)efficient for use in `PiiProcessor`. It is
+/// lossy in the sense that it cannot be consumed by downstream relays, so both versions have to be
+/// kept around.
+#[derive(Debug, Clone)]
+pub struct CompiledPiiConfig {
```

Comment:
We have quite some ownership problems here. The old `RuleRef` no longer actually is a reference, but an owned copy.. which also means any regex is cloned once after deserializing the PII config (but not for every event).